include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+    //!\brief Use alphabet container's size_type as a position.
+    //!\hideinitializer
+    using size_type = typename ranges::v3::size_type_t<inner_type>;
+    //!\}
+
+    /* rule of six */
+    /*!\name Constructors, destructor and assignment
+    * \{
+    */
+    // \brief Default constructor.
+    constexpr aligned_sequence_adaptor_constant_access() = default;
+
+    //!\brief Default copy constructor.
+    constexpr aligned_sequence_adaptor_constant_access(aligned_sequence_adaptor_constant_access const &) = default;
+
+    //!\brief Default copy construction via assignment.
+    constexpr aligned_sequence_adaptor_constant_access & operator=(aligned_sequence_adaptor_constant_access const &) = default;
+
+    /*!\brief Move constructor.
+    *
+    * The rank support structure of the bit vector needs the moved reference to
+    * the latter one. It has to be hand over subsequently. Therefore no default
+    * move assignment is possible. See https://github.com/xxsds/sdsl-lite/issues/15.
+    */
+    constexpr aligned_sequence_adaptor_constant_access (aligned_sequence_adaptor_constant_access && rhs) :
+    gap_vector(std::move(rhs.gap_vector)), sequence(std::move(rhs.sequence))
+    {
+        rs.set_vector(&gap_vector);
+        ss.set_vector(&gap_vector);
+    }
+
+    //!\brief Move assignment.
+    constexpr aligned_sequence_adaptor_constant_access & operator=(aligned_sequence_adaptor_constant_access && rhs)
+    {
+        gap_vector = std::move(rhs.gap_vector);
+        sequence = std::move(rhs.sequence);
+        rs.set_vector(&gap_vector);
+        ss.set_vector(&gap_vector);
+        return *this;
+    }
+
+    //!\brief Use default deconstructor.
+    ~aligned_sequence_adaptor_constant_access() = default;
+    //!\}
+
+    //!\brief
+    /*!\name Constructors of sequence concept.
+    * \{
+    */
+    //!\brief Construct by single value repeated 'size' times.
+    constexpr aligned_sequence_adaptor_constant_access(size_type size, value_type value)
+    {
+        assign(size, value);
+    };
+
+    //!\brief Construct by range given by two iterators of an aligned sequence.
+    constexpr aligned_sequence_adaptor_constant_access(iterator it1, iterator it2)
+    {
+        assign(it1, it2);
+    };
+
+    //!\brief Construct by sequence.
+    constexpr aligned_sequence_adaptor_constant_access(inner_type sequence_)
+    {
+        assign(sequence_);
+    };
+
+    //!\brief Construct by initializer_list for inner sequence container.
+    constexpr aligned_sequence_adaptor_constant_access(std::initializer_list<value_type> l)
+    {
+        assign(l);
+    };
+    //!\}
+
+    /*!\name Iterators
+     * \{
+    */
+    //!\brief Return iterator pointing to first element of underlying sequence.
+    auto begin() noexcept
+    {
+        return iterator{*this, 0};
+    }
+
+    //!\brief Return iterator pointing to past-the-end element of gapped sequence.
+    auto end() noexcept
+    {
+        return iterator{*this, size()};
+    }
```

Comment:
const versions of begin and end are missing (they are needed in addition to cbegin and cend!). I think you can just copy the iterator section from concatenated_sequences. You can also use \copydoc to copy the documentation entry from somewhere else if it is the same.